homeassistant/components/vizio/config_flow.py

```diff
@@ -198,7 +198,9 @@ def __init__(self) -> None:
 
             # Check if new config entry matches any existing config entries
             for entry in self.hass.config_entries.async_entries(DOMAIN):
-                if _host_is_same(entry.data[CONF_HOST], user_input[CONF_HOST]):
+                if CONF_HOST in entry.data and _host_is_same(
```

Comment:
Was there at any point a release where the config entry data didn't include the host key?

If not we shouldn't make this change. We shouldn't need to guard for file corruption in every config flow. If we want to guard for that, we need to figure out a general solution. Eg some validation during config entry loading.

Reply:
I can't recall a time when the host key was not required, because it wouldn't have made sense for it to be that way.

Re: validation during config entry loading, this may be challenging - I'm assuming it's a valid scenario for an integration to mutate input (user, import, or discovery) when creating the entry  data, so I think this may require a new schema provided by every integration that definitively defines what the entry data schema should look like. If, instead, each integration should handle this on their own, then I can look at adding validation logic into the setup process.

Reply:
If you are thinking that I should do the latter (handle this within this integration itself), what do you think should happen if the validation logic fails? Do we remove the entry?

Reply:
If we want validation on load, the validation should happen centrally, but the schema be provided by each integration. But that needs to be discussed in an architecture issue first.

I suggest we make sure to reproduce this issue and nail it down before doing anything more.

Reply:
👍